### ISSUE TITLE
Enhancement - Manage a DB of positions

### ISSUE THREAD

**Issue by GuidaMarco** [MuTsunTsai/fen-tool#12]
Hi.
I like a lot the tool and as soon as I discovered it, I immediately started using it on my iPhone.
Congratulations for your very good job!

While composing a chess problem it is often the case that you reach a position that is working, but not optimal. You would like therefore to "save" it and continue working to improve it. You might then reach an alternative position (say a "version" of the same problem) and you would like to save it as well, then continue working to further improve, possibly creating along the way several versions of the same problem. From time to time you might wish to quickly go from one version to another. It will be very useful to have the possibility to save (eg. locally on the device - iPhone in my case - so to work "off-line" with the tool) all such "versions" (alternative positions) of a given problem in a small local DB file, until your problem is finished. Then you might wish to compose another problem and create a new small DB of the position(s) you will then create along the way, and save it in another DB file.
 
A feature allowing to create and manage such DB file(s) (one for each "project"), and to allow to add, delete, etc. position(s) ("versions") in a given DB file will be very useful. To be clear, I am not asking for a fully-featured DB management, but only for a minimum set of DB features, strictly functional to support the creative and construction process of a chess problemist.

Thank you for your kind attention and looking forward to see future evolutions!
Best Regards
Marco Guida

**Comment by GuidaMarco** [comment]
Hi.

Just asking if there has been any progress on the enhancement proposed above ...
Thanks for your attention
Best Regards
Marco Guida

**Comment by MuTsunTsai** [comment]
@GuidaMarco Yes, I was busy with other matters a bit longer than expected, sorry about that. I might be able to work on it around next month, although I can't guarantee it.

**Comment by MuTsunTsai** [comment]
@GuidaMarco The basic functionality for creating projects is now done.

![image](https://github.com/MuTsunTsai/fen-tool/assets/39870664/d44bc89f-c9a8-4fa5-bb20-e45babc9e71c)

**Comment by GuidaMarco** [comment]
Hi.
Many thanks for your efforts. I checked it already on my iPhone and it seems OK.

I am not sure how it works the "New Project" Option: If I try to select it nothing happen. 
But I have been able to add positions, save a project; add other positions; save another project; open various projects; etc. It also works saving/opening both locally on iPhone and in the Cloud (with OneDrive). Wanderful!

Many thanks again for your work!
Ciao
Marco

**Comment by GuidaMarco** [comment]
Hi.
Yes, it works ... I did not understood the purpose / meaning; now it is clear!
Well done again!
Ciao
Marco
